pkg/ocrtypecodec/v1/exec.go

```diff
@@ -79,14 +79,20 @@ func (e *ExecCodecProto) DecodeObservation(data []byte) (exectypes.Observation,
 }
 
 func (e *ExecCodecProto) EncodeOutcome(outcome exectypes.Outcome) ([]byte, error) {
-	outcome = exectypes.NewSortedOutcome(outcome.State, outcome.CommitReports, outcome.Report)
+	outcome = exectypes.NewOutcome(outcome.State, outcome.CommitReports, outcome.Reports)
 
 	pbObs := &ocrtypecodecpb.ExecOutcome{
-		PluginState:   string(outcome.State),
-		CommitReports: e.tr.commitDataSliceToProto(outcome.CommitReports),
-		ExecutePluginReport: &ocrtypecodecpb.ExecutePluginReport{
-			ChainReports: e.tr.chainReportsToProto(outcome.Report.ChainReports),
-		},
+		PluginState:          string(outcome.State),
+		CommitReports:        e.tr.commitDataSliceToProto(outcome.CommitReports),
+		ExecutePluginReports: e.tr.execPluginReportsToProto(outcome.Reports),
+	}
+
+	// If there is only one report, use the legacy field. This way new clients can still
+	// form consensus with old ones.
+	// TODO: Remove backwards compatibility code after a few releases.
+	if len(pbObs.ExecutePluginReports) == 1 {
+		pbObs.ExecutePluginReport = pbObs.ExecutePluginReports[0]
+		pbObs.ExecutePluginReports = nil
```

Comment:
The encode function ensures that encoded exec outcomes leverage the legacy field unless multiple reports are detected. This should ensure new deployments produce outcomes that are identical to the previous version.